Cannot run tests on Ubuntu 14.04 LTS + Python 3

#### Repro Steps

```
# (Install Docker from: http://www.docker.com )

# If OS X, start with:
$ docker-machine start default; eval "$(docker-machine env default)"

# If OS X or Linux, continue with:
$ docker run -it ubuntu:trusty  # Ubuntu 14.04 LTS
$$ apt-get update
$$ apt-get install git -y
$$ apt-get install python3-pip -y
$$ pip3 install virtualenv
$$ git clone https://github.com/shazow/urllib3
$$ cd urllib3/
$$ virtualenv venv
$$ source venv/bin/activate
$$ make test
```

#### Output

```
python setup.py develop
Traceback (most recent call last):
  File "setup.py", line 23, in <module>
    long_description=open('README.rst').read() + '\n\n' + open('CHANGES.rst').read(),
  File "/urllib3/venv/lib/python3.4/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xf0 in position 4597: ordinal not in range(128)
make: *** [*.egg-info] Error 1
```

#### Notes

My best guess is that the `read()`s in setup.py should have an encoding specified. Probably UTF-8.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/32124743-cannot-run-tests-on-ubuntu-14-04-lts-python-3?utm_campaign=plugin&utm_content=tracker%2F192525&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F192525&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -192,6 +192,7 @@ In chronological order:
 
 * Jesse Shapiro <jesse@jesseshapiro.net>
   * Working on encoding unicode header parameter names
+  * Making setup.py resilient to ASCII locales
 
 * [Your name or handle] <[email or website]>
   * [Brief summary of your changes]

--- a/setup.py
+++ b/setup.py
@@ -4,7 +4,7 @@ from setuptools import setup
 
 import os
 import re
-
+import codecs
 
 base_path = os.path.dirname(__file__)
 
@@ -14,13 +14,14 @@ VERSION = re.compile(r".*__version__ = '(.*?)'",
                      re.S).match(fp.read()).group(1)
 fp.close()
 
-
+readme = codecs.open('README.rst', encoding='utf-8').read()
+changes = codecs.open('README.rst', encoding='utf-8').read()
 version = VERSION
 
 setup(name='urllib3',
       version=version,
       description="HTTP library with thread-safe connection pooling, file post, and more.",
-      long_description=open('README.rst').read() + '\n\n' + open('CHANGES.rst').read(),
+      long_description=u'\n\n'.join([readme, changes]),
       classifiers=[
           'Environment :: Web Environment',
           'Intended Audience :: Developers',

